src/CarbonAware.WebApi/test/unitTests/CarbonAwareControllerTests.cs

```diff
@@ -156,4 +156,31 @@ public void GetEmissions_NoLocations_ThrowsException(params string[] locations)
         Assert.ThrowsAsync<ArgumentException>(async () => await controller.GetEmissionsDataForLocationsByTime(locations));
         Assert.ThrowsAsync<ArgumentException>(async () => await controller.GetCurrentForecastData(locations));
     }
+
+    /// <summary>
+    /// Tests empty location arrays throw ArgumentException.
+    /// </summary>
+    [Test]
+    public async Task BatchForecast_NoLocations_ThrowsException()
+    {
+        var controller = new CarbonAwareController(this.MockCarbonAwareLogger.Object, CreateAggregatorWithEmissionsData(new List<EmissionsData>()).Object);
+        var forecastData = new List<EmissionsForecastBatchDTO>()
+        {
+            new EmissionsForecastBatchDTO
+            {
+                DataStartAt = new DateTimeOffset(2021,9,1,8,30,0, TimeSpan.Zero),
+                DataEndAt = new DateTimeOffset(2021,9,2,8,30,0, TimeSpan.Zero),
+                RequestedAt = new DateTimeOffset(2021,9,1,8,30,0, TimeSpan.Zero)
+            }
+        };
+        try
```

Comment:
Is there a reason we shouldn't use `Assert.ThrowsAsync<ArgumentException>` here instead of this block?

EG
`Assert.ThrowsAsync<ArgumentException>(async () => await controller.BatchForecastDataAsync(forecastData).ToList());`

Reply:
We cant use that syntax since this method returns a `IAsyncEnumerable` unlike the other methods in the Controller

Reply:
Refactored